nilearn/datasets/struct.py

```diff
@@ -106,38 +114,334 @@ def fetch_icbm152_2009(data_dir=None, url=None, resume=True, verbose=1):
     return Bunch(**params)
 
 
-def load_mni152_template():
-    """Load skullstripped 2mm version of the MNI152 originally distributed
-    with FSL.
+@deprecated("Default value of the resolution parameter in function "
```

Comment:
having 2 definitions of the function won't work. also we don't want to deprecate the use of the function itself.
the easiest way to warn about the default value is to make the new default `None`; like this:

```python
def load_mni152_template(resolution=None):
    if resolution is None:
        warnings.warn(
            "Default resolution will change from 2mm to 1mm in version x.y.z",
            FutureWarning)
        resolution = 2
    # ...
```

(note you could use some other sentinel value besides None so that the actual value appears in the signature, eg you could define some class so it looks like `def load_mni152_template(resolution=DefaultAboutToChange(2))`)